### PR TITLE
OS-8496 Deprecate docker registry access

### PR DESCRIPTION
**Pull request by bahamat**

### CODE REVIEW

File: src/img/lib/cli.js [hunk 1]
```diff
@@ -783,11 +783,6 @@ CLI.prototype.do_sources.options = [
         helpArg: '<source>',
         help: 'Add a source. It is appended to the list of sources.'
     },
-    {
-        names: ['add-docker-hub'],
-        type: 'bool',
-        help: 'A shortcut for "imgadm sources -t docker -a https://docker.io".'
-    },
```
Comment (danmcd) [inline src/img/lib/cli.js:790]: This doesn't have an sdc-imgapi counterpart?

**Comment by danmcd** [comment]
Also please add testing notes to the ticket.

File: src/img/lib/cli.js (hunk 1 above)
Comment (bahamat) [reply src/img/lib/cli.js:790]: Nope. This is purely for importing images into SmartOS.

**Comment by travispaul** [comment]
The `add-docker-hub` command is mentioned in a few other places including some tests in smartos-live. Do any of those references need to be updated/removed? 

https://hound.tritondatacenter.com/?q=add-docker-hub&i=nope&files=&repos=

**Review by danmcd — APPROVED** [review]
Need testing notes for IA.

**Review by danmcd — APPROVED** [review]

**Comment by danmcd** [comment]
Please make sure the node-imgadm gets its issues addressed as well.

**Comment by bahamat** [comment]
> The add-docker-hub command is mentioned in a few other places including some tests in smartos-live. Do any of those references need to be updated/removed?

The only place it's pertinent is in img/lib/cli.js and the man page. I'm only removing the documentation for it so that people will stop using it, but I'm not removing the feature. You can still use the `--add-docker-hub` flag, and add sources with type `docker`. In the future I might remove those as well, but we'll always need to handle the existence of docker sources, if only to ignore them.

**Review by danmcd — COMMENTED** [review]
Re-upping.

**Review by danmcd — APPROVED** [review]
Re-upping.

**Pull request merged**